recipes/funchook/all/test_package/conanfile.py

```diff
@@ -0,0 +1,32 @@
+import os
+from conans import ConanFile, CMake, tools
+
+
+class TestPackageConan(ConanFile):
+    settings = "os", "compiler", "build_type", "arch"
+    generators = "cmake", "cmake_find_package"
+
+    _cmake = None
+
+    def configure(self):
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.configure()
+        return self._cmake
+
+    def build(self):
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def test(self):
+        if not tools.cross_building(self):
+            # Funchook author reports hooking on x86_64 MacOS *can* work but only in shared libs
```

Comment:
if it does not work, it should not be distributed. You can use [validate](https://docs.conan.io/en/latest/reference/conanfile/methods.html#validate) to skip a package ID

Reply:
So, I don't have MacOS handy to test it, but my read from the Funchook documentation is that it *can* work, just not in all circumstances.  Hence why I felt it was better to simply skip my simple tests on MacOS.  The *real* way to fix this issue is to figure out how to make the tests work on MacOS but honestly that might be more motivation than I've got.

Any objection to continuing as-is (for this particular comment only)?

Reply:
You can not skip a test, you should skip the package generation if it does not work. Enable that configuration and wait for CI result, it will use Mac to build your package. If something goes wrong, you just can skip it.